dashboard/src/main/home/app-dashboard/validate-apply/metrics/MetricsSection.tsx

```diff
@@ -0,0 +1,352 @@
+import React, { useContext, useEffect, useState } from "react";
+import styled from "styled-components";
+
+import api from "shared/api";
+import { Context } from "shared/Context";
+import { ChartType } from "shared/types";
+
+import TabSelector from "components/TabSelector";
+import SelectRow from "components/form-components/SelectRow";
+import { getServiceNameFromControllerName, MetricNormalizer, resolutions, secondsBeforeNow } from "../../expanded-app/metrics/utils";
+import { Metric, MetricType, NginxStatusMetric } from "../../expanded-app/metrics/types";
+import { match } from "ts-pattern";
+import { AvailableMetrics, NormalizedMetricsData } from "main/home/cluster-dashboard/expanded-chart/metrics/types";
+import MetricsChart from "../../expanded-app/metrics/MetricsChart";
+import { useQuery } from "@tanstack/react-query";
+import Loading from "components/Loading";
+import CheckboxRow from "components/CheckboxRow";
+
+type PropsType = {
+    projectId: number;
+    clusterId: number;
+    appName: string;
+    services: Service[];
+    deploymentTargetId: string;
+};
+
+export type Autoscaling = {
+    minReplicas: number;
+    maxReplicas: number;
+    targetCPUUtilizationPercentage: number;
+    targetMemoryUtilizationPercentage: number;
+}
+
+export interface Service {
+    name: string;
+    kind: string;
+    ingress_enabled: boolean;
+    absolute_name?: string;
+    autoscaling?: Autoscaling;
+}
+
+type ServiceOption = {
+    label: string;
+    value: Service;
+}
+
+const MetricsSection: React.FunctionComponent<PropsType> = ({
+    projectId,
+    clusterId,
+    appName,
+    services,
+    deploymentTargetId,
+}) => {
+  const [selectedService, setSelectedService] = useState<Service>();
+  const [selectedRange, setSelectedRange] = useState("1H");
+  const [showAutoscalingThresholds, setShowAutoscalingThresholds] = useState(false);
+
+  const serviceOptions: ServiceOption[] = services.map((service) => {
+    return {
+      label: service.name,
+      value: service,
+    };
+  });
+
+    useEffect(() => {
+        if (services.length > 0) {
+            setSelectedService(services[0])
+        }
+    }, []);
+
+  const { data: metricsData, isLoading: isMetricsDataLoading, refetch } = useQuery(
+    [
+        "getMetrics",
+        projectId,
+        clusterId,
+        selectedService?.name,
+        selectedRange,
+    ],
+    async () => {
+      if (selectedService?.name == null) {
+        return;
+      }
+      const metrics: Metric[] = [];
+      const metricTypes: MetricType[] = ["cpu", "memory"];
+
+      const serviceName: string = selectedService.name
+      const isHpaEnabled: boolean = selectedService.autoscaling != null;
+
+      if (selectedService.kind === "web") {
+         metricTypes.push("network");
+      }
+
+      if (isHpaEnabled) {
+        metricTypes.push("hpa_replicas");
+      }
+
+      if (selectedService.ingress_enabled) {
+        metricTypes.push("nginx:status")
+      }
```

Comment:
I would separate out this logic into a useMemo and put it outside the useQuery fn:

```suggestion
const metricTypes = useMemo(() => {
      const metricTypes: MetricType[] = ["cpu", "memory"];

      const serviceName: string = selectedService.name
      const isHpaEnabled: boolean = selectedService.autoscaling != null;

      if (selectedService.kind === "web") {
         metricTypes.push("network");
      }

      if (isHpaEnabled) {
        metricTypes.push("hpa_replicas");
      }

      if (selectedService.ingress_enabled) {
        metricTypes.push("nginx:status")
      }
      
      return metricTypes
 }, [selectedService])
```